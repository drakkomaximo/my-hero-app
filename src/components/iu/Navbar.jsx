import React, { useContext } from "react";
import { Link, NavLink, useHistory } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContex";
import { linkRoutes } from "../../data/linkRoutes";
import { types } from "../../types/types";

export const Navbar = () => {
  const { heroUser, dispatch } = useContext(AuthContext);
  const history = useHistory()

  const handleLogout = () =>{

    dispatch({
      type: types.logout
    })

    history.replace('/login')

  }

  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">
        Asociaciones
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          {linkRoutes.map((linkRoute) => (
            <NavLink
              key={linkRoute.url}
              activeClassName="active"
              className="nav-item nav-link"
              exact
              to={linkRoute.url}
            >
              {linkRoute.label}
            </NavLink>
          ))}
        </div>
      </div>

      <div className="d-flex">
        <ul className="navbar-nav ml-auto">
          {heroUser.logged && (
            <span className="nav-item nav-link text-info">{heroUser.name}</span>
          )}
          <button 
            className="nav-item nav-link btn"
            onClick={ handleLogout }
            >Logout</button>
        </ul>
      </div>
    </nav>
  );
};
