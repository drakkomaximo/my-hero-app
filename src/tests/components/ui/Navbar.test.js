import { mount } from "enzyme"
import { MemoryRouter, Router } from "react-router"
import { AuthContext } from "../../../auth/AuthContex"
import { Navbar } from "../../../components/iu/Navbar"
import { types } from "../../../types/types"
import '@testing-library/jest-dom'

describe('Tests on <Navbar />', () => {

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        heroUser: {
            logged: true,
            name: 'María'
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    )

    afterEach(() => {
        jest.clearAllMocks()
    })

    test('It must show correctly', () => {
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.text-info').text().trim()).toBe('María')
    })

    test('It must call logout and use history', () => {

        wrapper.find('button').prop('onClick')()
        expect(contextValue.dispatch).toHaveBeenCalledTimes(1)
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        })
        expect( historyMock.replace ).toHaveBeenCalledWith('/login')

    })


})
