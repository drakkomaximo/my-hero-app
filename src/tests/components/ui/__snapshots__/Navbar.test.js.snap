// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Navbar /> It must show correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        BHA
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="d-flex"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Mar√≠a
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
